Tools/Scripts/libraries/webkitapipy/webkitapipy/macho.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (C) 2025 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import re
+import subprocess
+from pathlib import Path
+from typing import NamedTuple
+from enum import Enum
+import sys
+
+method_to_selector = re.compile(r'[-+]\[(?P<class>\S+) (?P<selector>[^\]]+)\]')
+
+
+class APIReport(NamedTuple):
+    file: Path
+    arch: str
+
+    exports: set[str] = set()
+    selectors: set[str] = set()
+    imports: set[str] = set()
+    selrefs: set[str] = set()
```

Comment:
Are these sets not held by the class, and not each instance? That seems… bad.

We probably need to override `__new__` to create new set instances for each NamedTuple; c.f. https://stackoverflow.com/a/59206181.

Also having sets (rather than frozensets) does imply the contents of the NamedTuple aren't actually constant.

Reply:
Oh, that's indeed bad! Switching to a dataclass is probably the better approach, where I can specify a [`default_factory`](https://docs.python.org/3.9/library/dataclasses.html#default-factory-functions) and we still have the autogenerated methods.